projects/components/src/table/table.component.ts

```diff
@@ -422,4 +443,31 @@ export class TableComponent
   public isRowExpanded(row: StatefulTableRow): boolean {
     return this.hasExpandableRows() && row.$$state.expanded;
   }
+
+  public onPageChange(pageEvent: PageEvent): void {
+    if (this.syncWithUrl) {
+      this.navigationService.addQueryParametersToUrl({
+        [TableComponent.PAGE_INDEX_URL_PARAM]: pageEvent.pageIndex,
+        [TableComponent.PAGE_SIZE_URL_PARAM]: pageEvent.pageSize
+      });
+    }
+  }
+
+  private initializeUrl(): void {
+    if (this.paginator) {
+      this.onPageChange({
+        pageSize: this.paginator.pageSize,
+        pageIndex: this.paginator.pageIndex
+      });
+    }
+  }
+
+  private pageDataFromUrl(params: ParamMap): Partial<PageEvent> | undefined {
+    return this.syncWithUrl
+      ? {
+          pageSize: new NumberCoercer().coerce(params.get(TableComponent.PAGE_SIZE_URL_PARAM)),
```

Comment:
nit: we could use the same coercer object.

Reply:
didn't seem worth it to me, felt It read cleaner this way rather than storing it in a var.